core/sql/generator/GenExplain.cpp

```diff
@@ -660,6 +660,269 @@ FileScan::addSpecificExplainInfo(ExplainTupleMaster *explainTuple,
+		{
+		  colFam[jj] = currPtr[currPos];
+		  currPos++;
+		  jj++;
+		}
+	      colFam[jj] = ':';
+	      jj++;
+	      currPos++;
+	      colFam[jj] = 0;
+	      colNameLen -= jj;
+
+	      NABoolean withAt = FALSE;
+	      char * colName = &currPtr[currPos];
+	      if (colName[0] == '@')
+		{
+		  colNameLen--;
+		  colName++;
+		  withAt = TRUE;
+		}
+
+	      Int64 v;
+	      if (colNameLen == sizeof(char))
+		v = *(char*)colName;
+	      else if (colNameLen == sizeof(unsigned short))
+		v = *(UInt16*)colName;
+	      else if (colNameLen == sizeof(Lng32))
+		v = *(ULng32*)colName;
+	      else
+		v = 0;
+	      if (j==0)
+		      str_sprintf(buf, "%s%s%Ld",
+				  colFam,
+				  (withAt ? "@" : ""),
+				  v);
+	      else
+			  str_sprintf(buf, ",%s%s%Ld",
+				  colFam,
+				  (withAt ? "@" : ""),
+				  v);
+
+	      outNAString += buf;
+
+	      listOfColNames->advance();
+	    } // for
+	}// trafodion tables
+	else
+	{// if hbase native tables
+	  char buf[1000];
+
+	  listOfColNames->position();
+	  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
+	    {
+	      char * currPtr = (char*)listOfColNames->getCurr();
+
+	      char * colNamePtr = NULL;
+
+		  Lng32 currPos = 0;
+		  short colNameLen = *(short*)currPtr;
+		  currPos += sizeof(short);
+		  char colName[500];
+
+		  for (Lng32 i = 0; i < colNameLen; i++)
+		    {
+		      colName[i] = currPtr[currPos];
+		      currPos++;
+		    }
+
+		  colName[colNameLen] = 0;
+
+		  colNamePtr = colName;
+
+		  if (j==0)
+			  str_sprintf(buf, "%s",colNamePtr);
+		  else
+			  str_sprintf(buf, ",%s",colNamePtr);
+
+
+		  outNAString += buf;
+
+	      listOfColNames->advance();
+	    } // for
+
+	}// hbase native table
+	outNAString +=" ";
+}
+
+static void appendPushedDownExpression(ComTdb *tdb, NAString& outNAString){
+	// in predicate pushdown V2, the hbaseCompareOps list contains a reverse polish set of operation, were operators are
+	// AND or OR, the rest are operands. this function display the column, operator and replace any constant with ?. it keeps reverse polish format
+	// this can be improved in the future for better readability.
+	char buf[1000];
+	Queue* reversePolishItems = ((ComTdbHbaseAccess *)tdb)->listOfHbaseCompareOps();
+	Queue* pushedDownColumns = ((ComTdbHbaseAccess *)tdb)->listOfHbaseFilterColNames();
+	reversePolishItems->position();
+	pushedDownColumns->position();
+
+	for (Lng32 j = 0; j < reversePolishItems->numEntries(); j++){
+		char * currPtr = (char*)reversePolishItems->getCurr();
+		char buf2[1000];
+		if (strcmp(currPtr,"V2")!=0 && strcmp(currPtr,"AND")!=0 && strcmp(currPtr,"OR")!=0){//if an operand (not an operator or V2 marker), get the column name
+			if (((ComTdbHbaseAccess*)tdb)->sqHbaseTable()){// if trafodion table
+				char * currPtr2 = (char*)pushedDownColumns->getCurr();
+			      Lng32 currPos = 0;
+			      Lng32 jj = 0;
+			      short colNameLen = *(short*)currPtr2;
+			      currPos += sizeof(short);
+			      char colFam[100];
```

Comment:
Same as above, limit taken from existing code. same 3 options:in ComTdbHbaseAccess.cpp:
static void showColNames(Queue * listOfColNames, Space * space)
{
  char buf[1000];

  listOfColNames->position();
  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
    {
      char * currPtr = (char*)listOfColNames->getCurr();
      
      Lng32 currPos = 0;
      Lng32 jj = 0;
      short colNameLen = *(short*)currPtr;
      currPos += sizeof(short);
      char colFam[100];
      while (currPtr[currPos] != ':')
	{
	  colFam[jj] = currPtr[currPos];
	  currPos++;
	  jj++;
	}
      colFam[jj] = ':';
      jj++;
      currPos++;
      colFam[jj] = 0;
      colNameLen -= jj;
      
      NABoolean withAt = FALSE;
      char * colName = &currPtr[currPos];
      if (colName[0] == '@')
	{
	  colNameLen--;
	  colName++;
	  withAt = TRUE;
	}
      
      Int64 v;
      if (colNameLen == sizeof(char))
	v = *(char*)colName;
      else if (colNameLen == sizeof(unsigned short))
	v = *(UInt16*)colName;
      else if (colNameLen == sizeof(Lng32))
	v = *(ULng32*)colName;
      else
	v = 0;
      
      str_sprintf(buf, "  Entry #%d: %s%s%Ld",
		  j+1, 
		  colFam, 
		  (withAt ? "@" : ""),
		  v);
      
      space->allocateAndCopyToAlignedSpace(buf, str_len(buf), sizeof(short));

      listOfColNames->advance();
    } // for
}